src/pages/studyView/StudyViewUtils.spec.ts

```diff
@@ -1303,4 +1305,75 @@ describe('StudyViewUtils', () => {
             assert.equal(getQValue(-0.01), '-1.000e-2');
         })
     });
+
+    describe('getSamplesByExcludingFiltersOnChart', () => {
+        let fetchStub: sinon.SinonStub;
+        const emptyStudyViewFilter: StudyViewFilter = {
+            clinicalDataEqualityFilters: [],
+            clinicalDataIntervalFilters: [],
+            cnaGenes: [],
+            mutatedGenes: []
+        } as any
+        beforeEach(() => {
+            fetchStub = sinon.stub(internalClient, 'fetchFilteredSamplesUsingPOST');
+            fetchStub
+                .returns(Promise.resolve([]));
+        });
+        afterEach(() => {
+            fetchStub.restore();
+        });
+
+        it('no filters selected', (done) => {
+            getSamplesByExcludingFiltersOnChart(
+                'WITH_MUTATION_DATA',
+                emptyStudyViewFilter,
+                {},
+                [{ sampleId: 'sample1', studyId: 'study1' }],
+                ['study1']
+            ).then(() => {
+                assert.isTrue(fetchStub.calledWith({ studyViewFilter: { ...emptyStudyViewFilter, sampleIdentifiers: [{ sampleId: 'sample1', studyId: 'study1' }] } }));
+                done();
+            }).catch(done);
+        });
+
+
+        it('has filter for one chart', (done) => {
+            getSamplesByExcludingFiltersOnChart(
+                'WITH_MUTATION_DATA',
+                emptyStudyViewFilter,
+                { 'WITH_MUTATION_DATA': [{ sampleId: 'sample1', studyId: 'study1' }], 'WITH_CNA_DATA': [{ sampleId: 'sample1', studyId: 'study1' }] },
+                [{ sampleId: 'sample1', studyId: 'study1' }, { sampleId: 'sample2', studyId: 'study1' }],
+                ['study1']
+            ).then(() => {
+                assert.isTrue(fetchStub.calledWith({ studyViewFilter: { ...emptyStudyViewFilter, sampleIdentifiers: [{ sampleId: 'sample1', studyId: 'study1' }] } }));
+                done();
+            }).catch(done);
```

Comment:
nice tests!  how come you want to catch though?  i think this would suppress a runtime error

Reply:
yes